Tests/NIOTLSTests/ApplicationProtocolNegotiationHandlerTests.swift

```diff
@@ -222,4 +243,37 @@ class ApplicationProtocolNegotiationHandlerTests: XCTestCase {
 
         XCTAssertTrue(try channel.finish().isClean)
     }
+
+    func testUnbufferingHandlesReentrantReads() throws {
+        let channel = EmbeddedChannel()
+        let loop = channel.eventLoop as! EmbeddedEventLoop
```

Comment:
This is just `let loop = channel.embeddedEventLoop` but looks like you never use the fact it's an embedded loop anyways:

```suggestion
        let loop = channel.eventLoop
```

Reply:
Yeah this was just a copy and paste from the old tests here. I fixed this up